File: POST_SERVICE/src/main/java/com/parkito/learnmicro/post/dto/UserDTO.java
package com.parkito.learnmicro.post.dto;

import lombok.Builder;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

/**
 * @author Artem Karnov @date 11/6/2017.
 * artem.karnov@t-systems.com
 */
@Builder
@Data
public class UserDTO implements Serializable {
    private String email;
    private String firstName;
    private String lastName;
    private List<String> serials;
}

File: USER_SERVICE/src/main/kotlin/com/parkito/learnmicro/user/service/dto/UserDTO.kt
package com.parkito.learnmicro.user.service.dto;

import com.parkito.learnmicro.user.service.entity.User
import java.io.Serializable
import java.util.*

/**
 * @author Artem Karnov @date 11/6/2017.
 * artem.karnov@t-systems.com
 */
data class UserDTO(
        val email: String = "",
        val firstName: String = "",
        val lastName: String = "",
        val serials: List<String> = Collections.emptyList()) : Serializable {

    companion object {
        fun fromEntity(entity: User, serials: List<String>) = UserDTO(
                email = entity.email,
                firstName = entity.firstName,
                lastName = entity.lastName,
                serials = serials
        )

        fun toEntity(userDTO: UserDTO) = User(
                email = userDTO.email,
                firstName = userDTO.firstName,
                lastName = userDTO.lastName
        )
    }
}
